konrad/atmosphere.py

```diff
@@ -362,7 +383,48 @@ def get_cold_point_plev(self, interpolate=False):
             return np.exp(-popt[1] / (2 * (popt[0])))
         else:
             # Return the single coldest point on the actual pressure grid.
-            return self['plev'][self.get_cold_point_index()]
+            return self["plev"][self.get_cold_point_index()]
+
+    def get_freezing_point_plev(self, interpolate=False):
+        """Return the cold point pressure.
+
+        Paramteres:
+            interpolate (bool): If `False` return the pressure grid value of
+                the closest point to T=0°C. If `True` perform a quadratic fit
+                to retrieve a smoother estimate of the freezing point pressure.
+
+        Returns:
+            float: Pressure at the freezing point [Pa].
+        """
+
+        if interpolate:
+            # Use the single coldest level between troposphere and stratosphere
+            # as starting point.
+            plog = np.log(self["plev"])
+            idx = self.get_freezing_point_index()
+
+            # Select a symmetric region [in ln(p)] around that level.
+            mask = np.logical_and(
+                plog > plog[idx] - 0.25,
+                plog <= plog[idx] + 0.25,
+            )
+
+            # Fit a linear polynomial to the temperature profile
+            # around the freezing point: f(x) = a*x + b
+            popt = np.polyfit(
+                plog[mask],
+                self["T"][-1, mask],
+                deg=1,
+            )
+
+            # Use a and b to determine the minmum of f(x) anayltically:
+            # f'(x) = 0
+            # 2 * a * x + b = 0
+            # x = -b / (2 * a)
+            return np.exp(273.15 / popt[0] - popt[1] / (popt[0]))
+        else:
+            # Return the single coldest point on the actual pressure grid.
+            return self["plev"][self.get_freezing_point_index()]
```

Comment:
This was probably only a misconception because of the naming but I would argue that this method is equivalent to `get_triple_point_plev()`

Reply:
I indeed missed the fact that triple point = freezing point, my bad. I removed both functions about freezing point and transfered the interpolation to the get_triple_point_plev function.

Reply:
I guess it is a good example on how being too specific can sometimes obfuscate what you want to say. The reasoning to use "triple point" is that it is a well defined temperature whereas for "freezing" we have other atmospheric conditions besides the temperature (e.g. the presence of aerosols or other condensation nuclei)